ros2service/ros2service/verb/call.py

```diff
@@ -44,23 +44,31 @@ def add_arguments(self, parser, cli_name):
             help='Values to fill the service request with in YAML format ' +
                  '(e.g. "{a: 1, b: 2}"), ' +
                  'otherwise the service request will be published with default values')
-
-        def float_or_int(input_string):
-            try:
-                value = float(input_string)
-            except ValueError:
-                raise argparse.ArgumentTypeError("'%s' is not an integer or float" % input_string)
-            return value
-
         parser.add_argument(
-            '-r', '--repeat', metavar='N', type=float_or_int,
-            help='Repeat the call every N seconds')
+            '-r', '--rate', metavar='N', type=float,
+            help='Calling rate (hz) default to 1')
+        parser.add_argument(
+            '-1', '--once', action='store_true',
+            help='Cal service once and exit')
 
     def main(self, *, args):
-        return requester(args.service_type, args.service_name, args.values, args.repeat)
-
-
-def requester(service_type, service_name, values, repeat):
+        return requester(args.service_type, args.service_name, args.values, args.rate, args.once)
+
+
+def requester(service_type, service_name, values, rate, once):
+    if rate is not None:
+        if once:
+            raise RuntimeError('You cannot select both -r and -1 (--once)')
+        try:
+            rate = float(rate)
+        except ValueError:
+            raise argparse.ArgumentTypeError('rate must be a number')
+        if rate <= 0:
+            raise ValueError('rate must be greater than zero')
+    if rate is not None:
+        period = 1. / rate
+    else:
+        period = 1
```

Comment:
I would suggest to use a default value for the argument instead. That way the code doesn't have to contain some not-documented constants.

That might affect the choice to group the arguments...

Reply:
Yeah I went back and forth on this.
I'm fine with adding the default vaIue for the rate and keeping the logic for the mutual exclusion. That logic can be in a `main` fuction before calling the actual requester like it's done for ros2topic. 

TBH I think that they dont have to be mutually exclusive. If you specify a rate you override a default of 1Hz, but if you specify `--once`, the rate (user specified or not) will not matter as we exit after the first call. I'm not convinced the executable should raise, it could just honor `once`